hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeDirectory.java

```diff
@@ -469,6 +475,7 @@ boolean addChild(final INode node, final boolean setModTime, final boolean
     if (logMetadataEvent) {
       node.logMetadataEvent(INodeMetadataLogEntry.Operation.Add);
     }
+    node.logProvenanceEvent(FileProvenanceEntry.Operation.create());
```

Comment:
you will need to handle the rename operation here, i guess just add a comment or fix me on top so we remember later that this is an issue here

Reply:
done